src/indexing.jl

```diff
@@ -60,20 +60,21 @@ end
 
 # ----- Base._checkbounds ----------------------------------------------------#
 
-function Base._checkbounds{T<:Real}(sz::Int, x::Nullable{T})
-    isnull(x) ? throw(NullException()) : Base._checkbounds(sz, get(x))
+function Base.checkbounds{T<:Real}(::Type{Bool}, sz::Int, x::Nullable{T})
+    isnull(x) ? throw(NullException()) : checkbounds(Bool, sz, get(x))
 end
 
-function Base._checkbounds(sz::Int, I::NullableVector{Bool})
-    length(I) == sz || throw(BoundsError())
+function Base.checkbounds(::Type{Bool}, sz::Int, I::NullableVector{Bool})
+    anynull(I) && throw(NullException())
+    length(I) == sz
```

Comment:
This changes the semantics here (beyond just the rename)… but I'm pretty sure that this is actually what you want.